### PR TITLE
Fix that skipSeat does not work if changed while turn is true

### LINKED ISSUE #1954: skipTurn does not clear out existing turn
If a seat has "turn:true" and then skipTurn is set to true, using the TURN function will not erase the "turn:true" and multiple seats then have the turn indicator.

### PR DESCRIPTION
**Pull request by 96LawDawg**
Fixes #1954

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: https://test.virtualtabletop.io/PR-1955/pr-test (or any other room on that server)

After merging, a backup will be available at `https://beta.virtualtabletop.io/editor/PR1955-pr-test`.

File: client/js/widgets/widget.js [hunk 1]
```diff
@@ -1833,8 +1833,9 @@ export class Widget extends StateManaged {
           problems.push(`Warning: turnCycle ${a.turnCycle} interpreted as forward.`);
           a.turnCycle = 'forward'
         }
-        let c = a.source === 'all' ? Array.from(widgets.values()) : collections[getCollection(a.source)] || [];
-        c = c.filter(w => w.get('type') === 'seat' && !w.get('skipTurn'));
+        let cBase = a.source === 'all' ? Array.from(widgets.values()) : collections[getCollection(a.source)] || [];
+        let c = cBase.filter(w => w.get('type') === 'seat' && !w.get('skipTurn'));
+        let cSkip = cBase.filter(w => w.get('type') === 'seat' && w.get('skipTurn'));
```
[the remark this answers is not in the document]
Comment (ArnoldSmith86) [reply client/js/widgets/widget.js:1838]: Oh. They aren't.

File: client/js/widgets/widget.js [hunk 2]
```diff
@@ -1888,6 +1889,11 @@ export class Widget extends StateManaged {
               collections[a.collection].push(w);
           }
 
+          //sets turn = false on any seats with skipTurn = true
+          for (const seat of cSkip) {
+            seat.set('turn', false);
```
[the remark this answers is not in the document]
Comment (96LawDawg) [reply client/js/widgets/widget.js]: Fixed.  Stupid ChatGPT.  I asked it about that and it confused me with the answer.  But it was probably right.

File: client/js/widgets/widget.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (96LawDawg) [reply client/js/widgets/widget.js:1838]: Yeah.  The TURN logic works on a collection of seats.  Then I made it a collection of seats without skipTurn.  So I needed a collection with skipTurn so I could turn that all off.  There may be another way, but that made sense to me.

**Review by ArnoldSmith86 — APPROVED** [review]

**Pull request merged**